### PR TITLE
DOC: Fix the references for `random.*`

### PR DESCRIPTION
**Pull request by takanori-pskq**
Related to #13114.

### CODE REVIEW

**Comment by mattip** [comment]
I think this [didn't work](https://16161-908607-gh.circle-artifacts.com/0/doc/build/html/release/1.17.0-notes.html#potential-changes-to-the-random-stream-in-old-random-module)

**Comment by takanori-pskq** [comment]
Fixed.
(At 90bbb1a, as you pointed the links are still broken, but sphinx emits no warnings...)

File: doc/source/release/1.17.0-notes.rst [hunk 1]
```diff
@@ -170,22 +170,18 @@ The functions `load`, and ``lib.format.read_array`` take an
...
-
 Potential changes to the random stream in old random module
 -----------------------------------------------------------
 Due to bugs in the application of ``log`` to random floating point numbers,
-the stream may change when sampling from `~RandomState.beta`, `~RandomState.binomial`,
-`~RandomState.laplace`, `~RandomState.logistic`, `~RandomState.logseries` or
-`~RandomState.multinomial` if a ``0`` is generated in the underlying `MT19937
-<~numpy.random.mt11937.MT19937>` random stream.  There is a ``1`` in
+the stream may change when sampling from `~numpy.random.RandomState.beta`,
+`~random.RandomState.binomial`, `~random.RandomState.laplace`,
```
Comment (eric-wieser) [inline doc/source/release/1.17.0-notes.rst:177]: Surely these need to match. I assume you want

```suggestion
the stream may change when sampling from `~random.RandomState.beta`,
`~random.RandomState.binomial`, `~random.RandomState.laplace`,
```

File: doc/source/release/1.17.0-notes.rst [hunk 2]
```diff
@@ -170,22 +170,18 @@ The functions `load`, and ``lib.format.read_array`` take an
 ``allow_pickle`` keyword which now defaults to ``False`` in response to
 `CVE-2019-6446 <https://nvd.nist.gov/vuln/detail/CVE-2019-6446>`_.
 
-
-.. currentmodule:: numpy.random.mtrand
```
Comment (eric-wieser) [inline doc/source/release/1.17.0-notes.rst:174]: Would changing this to `.. currentmodule:: numpy.random` have been enough to fix the links?

Comment (takanori-pskq) [reply doc/source/release/1.17.0-notes.rst:174]: That seems to be enough.

**Comment by rossbar** [comment]
Rebase + push to retrigger CI

**Review by rossbar — APPROVED** [review]
Looks like @eric-wieser 's suggestion did the trick! Will merge as soon as CI completes.

**Pull request merged**